app/src/main/kotlin/io/element/android/x/di/AppModule.kt

```diff
@@ -107,4 +109,10 @@ object AppModule {
     fun provideSnackbarDispatcher(): SnackbarDispatcher {
         return SnackbarDispatcher()
     }
+
+    @Provides
+    @SingleIn(AppScope::class)
```

Comment:
Use `ContributesBindings` in directly in the `AndroidClipboardHelper` constructor?

Reply:
`libraries:androidutils` doesn't have any DI, that's why I created it there instead. I can add Anvil support for the library instead if you think that's better.

Reply:
I think whenever we can it's easier to follow how it's used yes

Reply:
I have added `libraries:androidtools` module with DI to be able to do this.

Reply:
Well, I also added it 😅 . I guess we'll deal with the possible conflicts later.

Reply:
(but I will move my stuff to `libraries:androidutils` now).

Reply:
Ah, I didn't realise you meant `tools` and not `utils`,  my bad.